packages/umbra_flutter/test/src/umbra_exception_test.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:umbra_flutter/umbra_flutter.dart';
+
+class _MockTranspileException implements Exception {
+  const _MockTranspileException(this.op, this.message);
+
+  final int op;
+
+  final String message;
+
+  @override
+  String toString() => '$op: $message';
+}
+
+void main() {
+  group('UmbraException', () {
+    test('throw an exception when the initial message is unknown', () {
+      const exception = _MockTranspileException(0, 'unexpected');
+
+      expect(
+        () => UmbraException(exception),
+        throwsA(
+          isA<Exception>().having(
+            (e) => e.toString(),
+            'toString',
+            equals(
+              'Exception: Unknown transpiler exception message: unexpected',
+            ),
+          ),
+        ),
+      );
+    });
+
+    test('throw an exception when the error is not a TranspileException', () {
+      final exception = Exception('not a transpile one');
+
+      expect(
+        () => UmbraException(exception),
```

Comment:
It feels strange for the `UmbraException` factory to throw a different type of `Exception` imo.